apps/studio/src/features/dashboard/components/AdminCreateIndexPageButton.tsx

```diff
@@ -0,0 +1,81 @@
+import { Tooltip } from "@chakra-ui/react"
+import { Button, useToast } from "@opengovsg/design-system-react"
+import { BiLogoDevTo } from "react-icons/bi"
+
+import { BRIEF_TOAST_SETTINGS } from "~/constants/toast"
+import { useIsIsomerAdmin } from "~/features/permissions/hooks/useIsIsomerAdmin"
+import { trpc } from "~/utils/trpc"
+
+interface AdminCreateIndexPageButtonProps {
+  siteId: number
+  parentId: number
+}
+export const AdminCreateIndexPageButton = ({
+  siteId,
+  parentId,
+}: AdminCreateIndexPageButtonProps) => {
+  const toast = useToast()
+  const utils = trpc.useUtils()
+  const isIsomerAdmin = useIsIsomerAdmin()
+
+  const { data: indexPage } = trpc.resource.getIndexPage.useQuery({
+    siteId,
+    parentId: String(parentId),
+  })
+
+  const hasIndexPage = !!indexPage
+
+  const { mutate: createIndexPage, isLoading } =
+    trpc.page.createIndexPage.useMutation({
+      onSuccess: async () => {
+        // Invalidate only the necessary queries
+        await Promise.all([
+          utils.resource.getChildrenOf.invalidate({
+            siteId: String(siteId),
+            resourceId: String(parentId),
+          }),
+          utils.resource.listWithoutRoot.invalidate({ siteId: Number(siteId) }),
+          utils.collection.list.invalidate({ siteId: Number(siteId) }),
+          utils.resource.getIndexPage.invalidate({
+            siteId,
+            parentId: String(parentId),
+          }),
+        ])
+        toast({
+          status: "success",
+          title: "Index page created successfully",
+          ...BRIEF_TOAST_SETTINGS,
+        })
+      },
+      onError: (error) => {
+        toast({
+          status: "error",
+          title: "Failed to create index page",
+          description: error.message,
+          ...BRIEF_TOAST_SETTINGS,
+        })
+      },
+    })
+
+  if (!isIsomerAdmin) return null
```

Comment:
nit: should we be using `can` from `@casl/react` here? not a big deal but just for standardisation

Reply:
was considering that but feel like i didn't wanna overcomplicate it.
anyway have updated it to use growthbook feature flagging in https://github.com/opengovsg/isomer/pull/1065/commits/2f50412fb6e5c30f5cb88017285cf30b06bfc070 !